features/step_definitions/supplier_utils.rb

```diff
@@ -0,0 +1,138 @@
+#Utils
+
+# suppliers
+def create_supplier (supplier_id, supplier_name, supplier_description, supplier_contactName, supplier_email, supplier_postcode, supplier_dunsnumber)
+
+  file = File.read("./fixtures/test-supplier.json")
+  supplier_data = JSON.parse(file)
+  supplier_data["suppliers"]["id"] = supplier_id
+  supplier_data["suppliers"]["name"] = supplier_name
+  supplier_data["suppliers"]["description"] = supplier_description
+  supplier_data["suppliers"]["contactInformation"][0]["contactName"] = supplier_contactName
+  supplier_data["suppliers"]["contactInformation"][0]["email"] = supplier_email
+  supplier_data["suppliers"]["contactInformation"][0]["postcode"] = supplier_postcode
+  supplier_data["suppliers"]["dunsNumber"] = supplier_dunsnumber
+
+  supplier_path = "/suppliers/#{supplier_id}"
+
+  response = call_api(:get, supplier_path)
+  if response.code == 404
+    response = call_api(:put, supplier_path, payload: supplier_data)
+    response.code.should be(201), response.body
+  else
+    response = call_api(:post, supplier_path, payload: supplier_data)
+    response.code.should be(200), response.body
+
+    contact_id = JSON.parse(response.body)["suppliers"]["contactInformation"][0]["id"]
+    contact_data = JSON.parse("{\"updated_by\":\"Functional tests\", \"contactInformation\":#{supplier_data ["suppliers"]["contactInformation"][0].to_json}}")
+    response = call_api(:post, "/suppliers/#{supplier_id}/contact-information/#{contact_id}", payload: contact_data, safe_for_smoke_tests: true)
+    response.code.should be(200), response.body
+  end
+end
+
+# services
+def create_live_service (supplier_id, service_id, framework, lot)
```

Comment:
We have [this helper](https://github.com/alphagov/digitalmarketplace-functional-tests/blob/master/features/support/api_helpers.rb#L220) already for creating services which you may be able to extend to allow creation of g-cloud services (looks like it's only doing DOS at the moment.)